### ISSUE TITLE
Can't parse json

### ISSUE THREAD

**Issue by topilski** [json-c/json-c#343]
Hi next json: `{ "id": "encoding_122", "input_streams": [ { "id": 172, "last_pull_time": 1500819223, "bytes_in": 11394632, "in_bps": 223424, "is_broken": false } ], "output_streams": [ { "id": 81, "last_push_time": 1500819230, "bytes_out": 10412623, "out_bps": 201589, "is_broken": false } ], "uptime": 50, "rss": 431116, "cpu": 6.1089708310401765, "status_text": "Inited", "restarts": 0, "start_time": 1500819180, "cur_time": 1500819230 }` json_tokener_parse return NULL, why it happens?

Reproduced only on windows, MINGW compiler.

**Comment by hawicz** [comment]
No idea, it works just fine with a unix build.
Care to trim that down to a minimal test case so there's a chance of figuring out what's going wrong?  Or, perhaps simply using json_tokener_parse_ex and examining the error and state after the call will provide a clue.

**Comment by topilski** [comment]
Hi @hawicz, according debugger  in function ```
struct json_object* json_tokener_parse(const char *str)
{
    enum json_tokener_error jerr_ignored;
    struct json_object* obj;
    obj = json_tokener_parse_verbose(str, &jerr_ignored);
    return obj;
}
``` i see that :  jerr_ignored = json_tokener_error_parse_number (7) 
![image](https://user-images.githubusercontent.com/4686379/28741160-44a08808-7419-11e7-8a66-443c917dbdfb.png)

**Comment by topilski** [comment]
Implementation of function fix issue:
```
int json_parse_double(const char *buf, double *retval)
{
  char *end;
  *retval = strtod(buf, &end);
  return end == buf ? 1 : 0;
}
```

**Comment by hawicz** [comment]
Should be fixed in commit 65884f4d
